app/services/fastly_config/snippets.rb

```diff
@@ -0,0 +1,62 @@
+module FastlyConfig
+  class Snippets < Base
+    FASTLY_FILES = Rails.root.join("config/fastly/snippets/*.vcl").freeze
+
+    private
+
+    def file_updated?(filename)
+      snippet_name = File.basename(filename, ".vcl").humanize
+      snippet = find_snippet(fastly, active_version, snippet_name)
+
+      return true if snippet.nil?
+
+      new_snippet_content = File.read(filename)
+      new_snippet_content != snippet.content
+    end
+
+    def update_config(new_version, filename)
+      snippet_name = File.basename(filename, ".vcl").humanize
+      snippet = find_snippet(fastly, new_version, snippet_name)
+      new_snippet_content = File.read(filename)
+
+      if snippet
+        snippet.content = new_snippet_content
+        snippet.save!
+        log_to_datadog("update", snippet, new_version)
+      else
+        snippet_options = {
+          content: new_snippet_content,
+          dynamic: 0,
+          name: snippet_name,
+          service_id: ApplicationConfig["FASTLY_SERVICE_ID"],
+          type: "init",
+          version: new_version.number
+        }
+
+        snippet = fastly.create(Fastly::Snippet, snippet_options)
+        log_to_datadog("create", snippet, new_version)
+      end
```

Comment:
This is new behavior, find or create the snippet. If a community has Fastly configured, the scripts found in code will be updated on their Fastly.

Reply:
To make this a bit more obvious, how about calling the method `upsert_snippet`?

Reply:
I'd like to keep it generic because it's inherited from the `Base` class to be used in all configs. I changed it to `upsert_config`

Reply:
Good point and sensible change, thanks 👍